src/python/pants/core/goals/lint.py

```diff
@@ -170,6 +170,12 @@ def _requires_snapshot(cls) -> bool:
 
     @classproperty
     def tool_name(cls) -> str:
+        """The user-facing "name" of the tool."""
+        return cls.tool_subsystem.options_scope
+
+    @classproperty
+    def tool_slug(cls) -> str:
+        """The "slug" of the tool, used in tool selection (E.g. --only=<slug>)."""
```

Comment:
In some cases (listed in the PR) the name and slug differ. To give more color to those:
- For terraform the tool is `terraform` however the command is `terraform fmt`. Name should be `"terraform fmt"` but that's a terrible slug. In the context of `fmt` the command should read `--only=terraform`, since thats the tool.
- Likewise for `ruff`. `ruff <...>` does linting. `ruff --fix <...>` does fixing. name is `ruff --fix` in the fix case, and slug is always `ruff` in either case.

The intent is to capture these cases closer to the expectation of their selection.

Reply:
🤔 it's hard for me to fully grok the intended end state when only seeing part of it here.

As a user, I would be least-confused/least-surprised if the value printed in `fmt`/`fix`/`lint` summary output was identical to the value passed to `--only`. If we need to have some unique key to distinguish i.e. `ruff --fix` vs. `ruff` I _think_ I would expect that to be an internal implementation detail, not something shown to the user.

Reply:
Maybe we call a spade a spade? The names from `only` _should_ be `options_scope`s, since those are programmatically accessible.

BTW here's an example of the output from the upcoming `ruff` PR (notice the last two lines):

```
✕ autoflake failed.
✕ black failed.
✓ docformatter succeeded.
✕ flake8 failed.
✓ isort succeeded.
✓ preamble succeeded.
✕ ruff failed.
✕ ruff --fix failed.
```